src/client/containers/WelcomePage/WelcomePage.container.js

```diff
@@ -1,5 +1,53 @@
-import React from 'react';
+import React, { useState } from 'react';
+import './WelcomePage.styles.css';
+import GenericButton from '../../components/GenericButton/GenericButton.component';
+import GoogleButton from '../../components/GoogleButton/GoogleButton.component';
+import { useFirebase } from '../../firebase/FirebaseContext';
+import { Redirect } from 'react-router-dom';
 
 export default function Welcome() {
-  return <div>This is your private Welcome</div>;
+  const [redirect, setRedirect] = useState();
+  const { signIn, auth } = useFirebase();
+
+  function addUser() {
+    (async () => {
+      const token = await auth.currentUser.getIdToken();
+      await fetch('/api/users', {
+        method: 'POST',
+        mode: 'cors',
+        headers: {
+          'Content-type': 'application/json',
+          authorization: `Bearer ${token}`,
+        },
+      });
+    })();
+  }
+
+  const signInUser = async () => {
```

Comment:
We need to error handle addUser also.
Basically the user should only be redirected if the post request to create a user succeeded.

Reply:
I moved the redirect so it would take us to profile only if post request succeded, else it will stay on the same page and show the error on the console.